group_project_v2/stage/base.py

```diff
@@ -249,14 +266,41 @@ def get_dashboard_stage_state(self):
 
         return stage_state, state_stats
 
-    def get_stage_stats(self):  # pylint: disable=no-self-use
-        # TODO - real stats calculation
+    def get_stage_stats(self, target_workgroups, target_users):  # pylint: disable=no-self-use
+        target_user_ids = set(user.id for user in target_users)
+        if not target_user_ids:
+            return {
+                StageState.COMPLETED: 0,
+                StageState.INCOMPLETE: 0,
+                StageState.NOT_STARTED: 0
+            }
+
+        target_user_count = len(target_user_ids) * 1.0
+
+        completed_users, partially_completed_users = self.get_users_completion(target_workgroups, target_users)
+        log.info(STAGE_STATS_LOG_TPL.format(
+            stage=self.display_name,  target_users=[user.id for user in target_users],
```

Comment:
Nits:

Use `target_users=list(target_user_ids)` instead, perhaps?

Also, it's better to use the `log.info(STAGE_STATS_LOG_TPL, stage=self.display_name, ...)` format rather than calling `.format()` yourself, so that the string formatting is skipped when logging is disabled. The downside to that is you have to use old-style placeholders like `%(stage)s` instead of `{stage}` in your format string.